### PR TITLE
fix(dump): remove recursion

### PR DESCRIPTION
**Pull request by louis-bompart**
CDX-1117

## Proposed changes
image
This is often due to recursion that goes too far.
The only recursion in the search dump is the one removed in this PR.

## Testing

Corner case, so no automated tests.
`@coveo/cli@1.34.3-0` was released as alpha for validation on customer organization by support. waiting for confirmation from them.

### CODE REVIEW

File: packages/cli/src/commands/org/search/dump.ts [hunk 1]
```diff
@@ -279,19 +276,18 @@ export default class Dump extends Command {
...
-    const amountOfNewResultsToExtract =
-      maxAggregatedResults - this.aggregatedResults.length;
-    this.aggregatedResults.push(
-      ...newResults.splice(
-        0,
-        Math.min(amountOfNewResultsToExtract, newResults.length)
-      )
-    );
-    if (newResults.length > 0) {
+    while (results.length > 0) {
```
Comment (louis-bompart) [inline packages/cli/src/commands/org/search/dump.ts:281]: We can refactor like this because the splice was already noop if `results.length===0` :
1. `Math.min(amountOfNewResultsToExtract, results.length) == 0`
2. `[].splice(0,0) == []`
3. array.push(...[]) does nothing


However, that makes the method dumping the result no matter what in the end.
This was first made by the other chunk that was removed above.
I checked the order of operations and I don't think that it would change anything.

**Comment by louis-bompart** [comment]
- [ ] Provide Jira #
- [ ] Support validated that the alpha version fix the issue

**Review by y-lakhdar — APPROVED** [review]
